File: leetsync/2891-maximum-beauty-of-an-array-after-applying-operation/maximum-beauty-of-an-array-after-applying-operation.java
class Solution {
    public int maximumBeauty(int[] nums, int k) {
        int maxBeauty = 0;
        Arrays.sort(nums);
        int n = nums.length;
        int i=0, j=0;
        // Sliding Window
        while(i<n){
            while(j<n && nums[j]-nums[i]<=2*k){
                j++;
            }
            maxBeauty = Math.max(maxBeauty, j-i);
            i++;
        }
        return maxBeauty;
    }
}

File: leetsync/2891-maximum-beauty-of-an-array-after-applying-operation/maximum-beauty-of-an-array-after-applying-operation.go
package main

import (
	"sort"
)

func maximumBeauty(nums []int, k int) int {
	maxBeauty, i, j := 0, 0, 0
	sort.Ints(nums)
	n := len(nums)
	for i < n {
		for j < n && ((nums[j] - nums[i]) <= (2 * k)) {
			j++
		}
		if maxBeauty < (j - i) {
			maxBeauty = j - i
		}
		i++
	}
	return maxBeauty
}
